client/js/s3/request-signer.js

```diff
@@ -153,52 +154,18 @@ qq.s3.RequestSigner = function(o) {
             },
 
             getEncodedHashedPayload: function(body) {
-                var promise = new qq.Promise(),
-                    reader,
-                    self = this;
-
+                var promise,
+                    reader;
                 if (qq.isBlob(body)) {
-                    // Only if a worker is requested will we load it.
-                    // We will fallback to the inline reader if the worker
-                    // fails to load.
-                    if (options.signatureSpec.useWorker) {
-                        if (!this._worker) {
-                            try {
-                                // allow the worker path to be overwritten.. Eg. we aren't using an inline worker.
-                                var workerUrl = options.signatureSpec.useWorker;
-                                if (workerUrl === true) {
-                                    if (!qq.s3.worker) {
-                                        throw new Error("Missing inline s3 worker.");
-                                    }
-                                    workerUrl = qq.s3.worker();
-                                }
-                                this._worker = new Worker(workerUrl);
-                                this._workerPromises = {};
-                                this._worker.onmessage = function (e) {
-                                    if (!self._workerPromises[e.data.id]) {
-                                        options.log("Worker returned a result for an request we dont know about.");
-                                        return;
-                                    }
-                                    if (e.data.err) {
-                                        self._workerPromises[e.data.id].failure(e.data.err);
-                                    } else {
-                                        self._workerPromises[e.data.id].success(e.data.resp);
-                                    }
-                                    delete self._workerPromises[e.data.id];
-                                };
-                            } catch (ex) {
-                                // worker is not supported or invalid
-                                options.log("Worker failed to be created. Defaulting back to main thread processing.", ex);
-                                options.signatureSpec.useWorker = false;
-                            }
-                        }
-                        if (this._worker) {
-                            var task = {file: body, id: qq.getUniqueId()};
-                            this._workerPromises[task.id] = promise;
-                            this._worker.postMessage(task);
+                    // We will fallback to the inline reader if the worker was
+                    // not loaded correctly
+                    if (workerManager) {
+                        promise = workerManager.generateSignature(body);
+                        if (promise !== null) {
```

Comment:
Is it possible that `promise` can be `undefined` here? If so, this should be `if (promise != null {`.